bottlecap/src/otlp/transform.rs

```diff
@@ -0,0 +1,1144 @@
+use datadog_trace_normalization::normalize_utils::{normalize_service, normalize_tag};
+use datadog_trace_protobuf::pb::Span as DatadogSpan;
+use hex;
+use lazy_static::lazy_static;
+use opentelemetry_proto::tonic::common::v1::{
+    any_value, AnyValue, InstrumentationScope as OtelInstrumentationScope, KeyValue,
+};
+use opentelemetry_proto::tonic::resource::v1::Resource as OtelResource;
+use opentelemetry_proto::tonic::trace::v1::ResourceSpans;
+use opentelemetry_proto::tonic::trace::v1::{
+    span::{Event, Link, SpanKind},
+    status::StatusCode,
+    Span as OtelSpan,
+};
+use opentelemetry_semantic_conventions::attribute::{
+    DB_QUERY_TEXT, DB_STATEMENT, DB_SYSTEM_NAME, DEPLOYMENT_ENVIRONMENT, FAAS_INVOKED_NAME,
+    FAAS_INVOKED_PROVIDER, FAAS_TRIGGER, GRAPHQL_OPERATION_NAME, GRAPHQL_OPERATION_TYPE,
+    HTTP_ROUTE, HTTP_STATUS_CODE, MESSAGING_DESTINATION_NAME, MESSAGING_OPERATION,
+    MESSAGING_SYSTEM, OTEL_LIBRARY_NAME, OTEL_LIBRARY_VERSION, OTEL_STATUS_CODE,
+    OTEL_STATUS_DESCRIPTION, RPC_METHOD, RPC_SERVICE, RPC_SYSTEM,
+};
+use opentelemetry_semantic_conventions::resource::SERVICE_VERSION; // CONTAINER_ID, SERVICE_VERSION, TELEMETRY_SDK_LANGUAGE, TELEMETRY_SDK_VERSION,
+use opentelemetry_semantic_conventions::trace::{
+    EXCEPTION_MESSAGE, EXCEPTION_STACKTRACE, EXCEPTION_TYPE,
+};
+use serde_json::json;
+use std::collections::HashMap;
+use std::str;
+use std::sync::Arc;
+use tracing::debug;
+
+use crate::config::Config;
+use crate::traces::propagation::text_map_propagator::DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY;
+
+// Constants for Datadog namespace keys
+pub const KEY_DATADOG_SERVICE: &str = "datadog.service";
```

Comment:
nit: these could all be constants in a `constants.rs` if you wanted to make this file less large

Reply:
I want to leave them here as I will eventually merge this from a crate from the tracer